### PR TITLE
Fix comment syntax inside jsx

### PR DESCRIPTION
**Pull request by goodmind**
Temporary workaround for #7672

```js
function Component<T>(props: {|type: T|}) {
  return null
}

const el = <Component /*::type={1}*/ />
```

### CODE REVIEW

**Comment by gabelevi** [comment]
What's the use case for this? Has this been requested?

**Comment by goodmind** [comment]
@gabelevi isn't it more consistent for comment syntax to work everywhere?
This is workaround for providing generic type to component
```js
function Component<T>(props: {|type: T|}) {}
const el = <Component /*::type={1}*/ />
```

**Comment by goodmind** [comment]
@gabelevi I updated main post

File: src/parser/flow_lexer.ml [hunk 1]
```diff
@@ -1501,6 +1501,40 @@ let jsx_tag env lexbuf =
...
+        Continue env
+
+  | "*/" ->
+    if is_in_comment_syntax env then
+      let env = in_comment_syntax false env in
+      Continue env
+    else begin
+      Sedlexing.rollback lexbuf;
+      match%sedlex lexbuf with
+      | "*" -> Token (env, T_MULT)
```
Comment (gabelevi) [inline src/parser/flow_lexer.ml]: This isn't right. Outside of a JSX tag, `*` would be the `T_MULT` token (for multiplication). But inside of a JSX tag I think it's an unexpected token. You can probably just emit an error. Something like

```
| "*" -> Token (env, T_ERROR (lexeme lexbuf))
```

**Comment by goodmind** [comment]
I can't test it now, but I'm not sure if this would be allowed too?
```js
</*::Test*/ />
```

**Comment by goodmind** [comment]
@gabelevi I'm going to close it because I'm not really want to introduce some unexpected behaviour here with jsx, also I think this is not worth it

**Pull request closed without merging**